### PR TITLE
fix: incorrect context timeout during listPath()

### LINKED ISSUE #15405: Deletion of objects too slow on single drive setup
## NOTE
Recently we have upgraded to version 2022-07-26. Since that we have problems with deletion objects from buckets. This may take many hours.
Have to notice, we use one node setup and new format of data-catalogs for data which are support versioning. A bucket is created with disabled versioning and other features.

## Expected Behavior
When S3 API is using: server return code 200 after total files deletion.
When GUI is using: the deletion progress can be observed and the operation can be performed in synchronous mode. 

## Current Behavior
When a deletion operation performs with using S3 API, file removing from drive too slowly. 
For example, if I put to a bucket many files (about 150Gb) and perform its deletion, it could take about hour. Its deleteing with using Minio GUI also don't work, it shows a message with timeout error. 
image

## Possible Solution
Some flag to perform delete operation in synchronous mode.

## Steps to Reproduce (for bugs)
1. Use Minio 2022-07-26;
2. Create new data-catalog
3. Create new bucket with disabled all features;
4. Create some catalogue and fill it with a large number of files, the total size should be more than 100Gb.
5. Try to remote the catalogue. 

## Context
I am using S3 to store my backups. Before uploading a new backup to S3, I want to delete the old one to free up some space. Now I sometimes run out of free space because files are being deleted too slowly.

## Regression
With previous version (2022-06...) we didn't have such problems. 

## Your Environment
* Version used (`minio --version`): RELEASE.2022-06-25T15-50-16Z (commit-id=bd099f5e71d0ea511846372869bfcb280a5da2f6)
* Server setup and configuration: 8vCPU, 16Gb RAM, RAID 5 with 5 disks. Using Single-Node Single-Drive setup with data-directory on XFS volume.
* Operating System and version (`uname -a`): Centos 7, Core: Linux 3.10.0-1160.49.1.el7.x86_64 #1 SMP x86_64 x86_64 x86_64 GNU/Linux

### LINKED ISSUE #15442: A timeout occurred while trying to lock a resource, please reduce your request rate
## NOTE
This problem is encountered after I open HTTPS, when I try to delete files with .exe and .apk extensions, they will not be deleted and the message "Timeout occurred while trying to lock the resource, please reduce your request rate" content
Currently this problem I found only in the .exe and .apk suffix files, when I delete the zip format or image format is all normal, upload any suffix file is also all normal, only in the deletion of these two suffixes encountered (there may be other suffix files)
 

## Expected Behavior
Under normal circumstances, the file will be deleted

## Current Behavior
Prompt "A timeout occurred while trying to lock a resource, please reduce your request rate"
HTTPS URL /api/v1/buckets/test/delete-objects?all_versions=false
The returned json is
```json
{"code":500,"detailedMessage":"A timeout occurred while trying to lock a resource, please reduce your request rate","message":"an error occurred, please try again"}
```

## Steps to Reproduce (for bugs)

1. Upload the crt certificate and key file of the domain to the /root/.minio/certs directory of the server
2. My startup command is
/minio/minio server /mnt/sda1 --console-address ":2083" --address "domain.com:2053"
3. Modify the hosts file and add the following content
192.168.31.2 domain.com
If not added, my minio can not start, this is the way I found in Google
4. Upload xx.exe and xx.apk files to the storage bucket
5. Error deleting xx.exe and xx.apk

## Context

The server is Linux arm64
I started minio using the binary
The version of minio is the latest version as of July 30, 2022

## Regression

I did not try to roll back the version, but the problem persists in versions after at least July 26, 2022

## Your Environment

* Version used (`minio --version`):
```
minio version RELEASE.2022-07-30T05-21-40Z
```
* Operating System and version (`uname -a`):
```
Linux OpenWrt 5.4.203 #0 SMP PREEMPT Tue Jul 12 10:40:55 2022 aarch64 GNU/Linux
```

### LINKED ISSUE #15508: "A timeout occurred while trying to lock a resource, please reduce your request rate" when trying to list objects
## NOTE
code:
```
public void test() throws Exception {
        log.info("start test object list");
        long start = System.currentTimeMillis();
        int count = this.listAll("");
        log.info("test object list complete, spend: {}s, count: {}", (System.currentTimeMillis() - start) / 1000, count);
    }

    private int listAll(String prefix) {
        int count = 0;
        Iterable<Result<Item>> parts = minioClient.listObjects(ListObjectsArgs.builder()
                .prefix(prefix)
                .bucket("test")
                .recursive(true)
                .build());
        Iterator<Result<Item>> iterator = parts.iterator();
        while (iterator.hasNext()) {
            try {
                iterator.next().get();
                count++;
            } catch (Exception e) {
                log.info("list object error: {}", e.getMessage(), e);
            }
        }
        return count;
    }
```
console log:
```
2022-08-09 21:26:03.047  INFO 68233 --- [   scheduling-1] com.service.TestService            : list object error: A timeout occurred while trying to lock a resource, please reduce your request rate

io.minio.errors.ErrorResponseException: A timeout occurred while trying to lock a resource, please reduce your request rate
	at io.minio.S3Base$1.onResponse(S3Base.java:690) ~[minio-8.4.2.jar:8.4.2]
	at okhttp3.internal.connection.RealCall$AsyncCall.run(RealCall.kt:519) ~[okhttp-4.10.0.jar:na]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) ~[na:1.8.0_312]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) ~[na:1.8.0_312]
	at java.lang.Thread.run(Thread.java:748) ~[na:1.8.0_312]

2022-08-09 21:26:03.047  INFO 68233 --- [   scheduling-1] com.service.TestService            : test object list complete, spend: 10s, count: 2000
```

## Your Environment

* Version used (`minio --version`): RELEASE.2022-08-05T23-27-09Z
* Server setup and configuration: 
* Operating System and version (`uname -a`): ubuntu18.04 4.15.0-180-generic

### PR DESCRIPTION
**Pull request by harshavardhana**
## Description
fix: incorrect context timeout during listPath()

## Motivation and Context
This PR cleans up the listing code for single drive
to ensure that we do not add an incorrect context
timeout, while resuming the listing.

fixes #15508
fixes #15442
fixes #15405

## How to test this PR?
Have a directory that takes > 5secs to respond for WalkDir() 
so it must be a sizable folder with lots of content. 

## Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Optimization (provides speedup with no functional changes)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [x] Fixes a regression This is a bug from the single drive XL setup implementation, so a regression in some sense.
- [ ] Documentation updated
- [ ] Unit tests added/updated

### CODE REVIEW

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-erasure.sh | :heavy_check_mark: |
| mint-dist-erasure.sh | :heavy_check_mark: |
| mint-compress-encrypt-dist-erasure.sh | :heavy_check_mark: |
| mint-pools.sh | :heavy_check_mark: |
Deleting image on docker hub
Deleting image locally

File: cmd/metacache-server-pool.go [hunk 1]
```diff
@@ -318,58 +318,32 @@ func (es *erasureSingle) listPath(ctx context.Context, o *listPathOptions) (entr
 	//
 	// If we don't have a list id we must ask the server if it has a cache or create a new.
 	if o.ID != "" && !o.Transient {
-		// Create or ping with handout...
-		rpc := globalNotificationSys.restClientFromHash(pathJoin(o.Bucket, o.Prefix))
```
Comment (vadmeste) [inline cmd/metacache-server-pool.go:322]: I am not sure why this is removed, this means we are not going to use list caching anymore?

Comment (vadmeste) [reply cmd/metacache-server-pool.go:322]: > I am not sure why this is removed, this means we are not going to use list caching anymore?

Never mind, just found out this is erasure single

**Review by vadmeste — APPROVED** [review]
LGTM

**Pull request merged**